File: targets/numworks/tests/camlboot/benchs/object.ml
class point x y = object
  val x = x
  val y = y
  method sym = (new point (-x) (-y))
end

let _ =
  let start = millis () in
  for _ = 1 to 100 do (* x 1000 to compare with OMicroB *)
        let o = new point 10 10 in
        for _ = 1 to 10 do
          let _ = o#sym in ()
        done
  done;
  let stop = millis () in
  print_int (stop-start)

File: targets/numworks/tests/camlboot/benchs/object.py
import time

class Point:
    def __init__(self, x,y):
        self.x = x
        self.y = y
    
    def sym(self):
        return (Point(-self.x,-self.y))

start = time.monotonic()
for i in range(0,100):
    o = Point(10,10)
    for k in range (0,10):
        o.sym()
stop=time.monotonic()
print(stop-start)
